integration-tests/trace/llama_index/test_callback.py

```diff
@@ -33,6 +38,10 @@
     MESSAGE_ROLE,
     OUTPUT_MIME_TYPE,
     OUTPUT_VALUE,
+    RERANKING_INPUT_DOCUMENTS,
+    RERANKING_MODEL_NAME,
+    RERANKING_OUTPUT_DOCUMENTS,
+    RERANKING_TOP_K,
```

Comment:
quick question on this parameter: https://docs.cohere.com/docs/reranking

I'm guessing this is the same as TOP_N? If you feed say 5 documents but pass top_k of 3, does it only rank 3? Just trying to understand why this is a parameter to the rerank model.

Reply:
Yes, this is the same as TOP_N. (The letter is `K` in [literature](https://nlp.stanford.edu/IR-book/html/htmledition/inexact-top-k-document-retrieval-1.html) because `N` is usually the total number of docs.) The caller of the reranker usually just wants a relatively small number of docs out of a initial set of tens or hundreds. It's certainly optional because it can just rank each document, but in general, a reduction in number is expected for each stage of the retrieval process.

Reply:
Still confused though - if I pass say 5 documents with `top_k` of 3 - does it rank 5 and trim the last two?

Reply:
Yes, it's retains up to `K` in the output, so in this case 2 (of the lowest scores) has to be dropped. `Top K` is applied after ranking all 5 docs.